docs/docs/conceptual/rendering-options.md

```diff
@@ -0,0 +1,93 @@
+other rendering options are available: Deferred Static Generation (DSG) and Server-side rendering (SSR).
+Let's dive into what each of those modes mean.
+
+## Static Site Generation (SSG)
+
+SSG is the default rendering mode in Gatsby. Gatsby generates all assets and HTML for all pages at build time on a
+build server (this could be your laptop, any build service, or Gatsby Cloud worker if you use [Gatsby Cloud][6]).
+
+Then, the produced static files are uploaded to the CDN provider of choice and served to end-users.
+The build server is not needed after the build step and could even be turned off.
+
+This mode provides the most pleasant user experience, the highest level of security, and run-time scalability for your site.
+
+> **Note:** SSG doesn't mean your site is not dynamic. You can still use JavaScript to communicate with any APIs,
+> add private sections of your site for authorized users via [client-side rendering][4] and
+> have any features SPA can have
+
+Here is how SSG works in Gatsby Cloud (the principle remains the same with any build and CDN provider):
+
+![Static Site Generation diagram](../images/ssg-diagram.jpg)
+
+One downside of SSG is building time. As the number of pages of your site grows, so does the build time.
+Gatsby supports [incremental builds][5] to make sure the 2nd and subsequent builds only rebuild what was changed
+but for the initial build (the build without the cache), build times may become an issue. That's where the
+Deferred Static Generation could be beneficial.
+
+## Deferred Static Generation (DSG)
+
+As the title suggests, Deferred Static Generation is conceptually very similar to SSG. The only difference is that some
+pages are not generated at build time and instead generated during the first HTTP request to this page.
+
+The subsequent requests will hit the CDN cache the same way as with SSG.
+
+For example: maybe you have an archive of old articles that no longer receive significant traffic. There is
+no practical reason to generate them on each build and thus delay the delivery of fresh articles.
+
+In this case, you may choose to defer the generation of old pages, and Gatsby will skip them during the build step.
+
+- [How to use deferred static generation][7]
+
+![Deferred Static Generation diagram](../images/dsg-diagram.jpg)
+
+Unlike SSG, deferred static generation requires you to keep the build server running after the initial build.
+It implies a different deployment model and requires backend infrastructure.
+
+But don't worry, [Gatsby Cloud][6] supports it out-of-the-box.
+
+## Server-side Rendering (SSR)
+
+SSG, DSG, and client-side rendering can handle a vast majority of use-cases in web development.
+
+But there is a small niche when you may still need to generate HTML on-the-fly.
+
+For example: maybe you are building a site with user reviews and want those reviews to be
+immediately indexed by search engines as soon as they are posted. So the client-side rendering is not an option.
```

Comment:
bit rusty on this, but don't search engines (particularly google) will actually use js, so indexing still would happen?

Reply:
It is being indexed but JS based sites are slower

Reply:
Perhaps we should word it to something along these lines:

```
For example: you have a personalised landing page for each user. Getting the data at server time can reduce the page from shifting its layout while data gets loaded. Therefore, it can give a better user experience than client side only rendering.

Reply:
Yeah, according to [google docs](https://developers.google.com/search/docs/advanced/javascript/javascript-seo-basics) google bot does run JS. But not sure if it is the same for all search engines (Bing, Yandex, etc).

But even those docs contain the following note:

> Keep in mind that server-side or pre-rendering is still a great idea because it makes your website faster for users and crawlers, and not all bots can run JavaScript.